### PR TITLE
feat(db): record client version history

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

h/t @joshieDo for the idea

Record client version history on database access with write privileges.

### CODE REVIEW

File: crates/storage/db/src/tables/models/client_version.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    fn from_compact(buf: &[u8], len: usize) -> (Self, &[u8]) {
+        let (version, buf) = Vec::<u8>::from_compact(buf, len);
+        let (git_sha, buf) = Vec::<u8>::from_compact(buf, len);
+        let (build_timestamp, buf) = Vec::<u8>::from_compact(buf, len);
+        let client_version = Self {
+            version: unsafe { String::from_utf8_unchecked(version) },
+            git_sha: unsafe { String::from_utf8_unchecked(git_sha) },
+            build_timestamp: unsafe { String::from_utf8_unchecked(build_timestamp) },
```
Comment (rkrasiuk) [inline crates/storage/db/src/tables/models/client_version.rs:43]: open to making it safe

File: crates/storage/db/src/tables/models/client_version.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#[derive_arbitrary(compact)]
+#[derive(Clone, Eq, PartialEq, Debug, Default, Serialize, Deserialize)]
+pub struct ClientVersion {
+    /// Client version
+    pub version: String,
+    /// The git commit sha
+    pub git_sha: String,
+    /// Build timestamp
+    pub build_timestamp: String,
+}
```
Comment (shekhirin) [inline crates/storage/db/src/tables/models/client_version.rs:16]: can we just do `#[main_codec]` here?

Comment (rkrasiuk) [reply crates/storage/db/src/tables/models/client_version.rs:16]: it complains about strings

Comment (shekhirin) [reply crates/storage/db/src/tables/models/client_version.rs:16]: sad but true

**Review by joshieDo — APPROVED** [review]

**Pull request merged**